lib/src/presentation/movie_list_view.dart

```diff
@@ -28,106 +24,109 @@ class MovieListView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final categories = repository.moviesList();
+    final categoryNames = [
+      "Ação",
+      "Animação",
+      "Drama",
+      "Comédia",
+      "Sci-Fi",
+      "Crime"
+    ];
 
     return Scaffold(
       appBar: AppBar(
         centerTitle: true,
         title: SizedBox(
-          height: AppBar().preferredSize.height*0.7,
+          height: AppBar().preferredSize.height * 0.7,
           child: Image(
-            image: checkThemeMode(context),),
+            image: checkThemeMode(context),
+          ),
         ),
-
-          actions: [
+        actions: [
           IconButton(
             icon: const Icon(Icons.settings),
             onPressed: () {
               Navigator.push(
                 context,
                 MaterialPageRoute(
-                  builder: (context) => SettingsView(controller: settingsController),
+                  builder: (context) =>
+                      SettingsView(controller: settingsController),
                 ),
               );
             },
           ),
         ],
       ),
-      body: _buildBody(),
+      body: _buildBody(categories, categoryNames),
       bottomNavigationBar: _buildBottomNavigationBar(),
     );
   }
 
-  Widget _buildBody() {
-    return SafeArea(
-      child: ListView(
-        children: [
-          _buildCategoryListView('Ação', items),
-          _buildCategoryListView('Comédia', items),
-          _buildCategoryListView('Drama', items),
-          _buildCategoryListView('Animação', items),
-          // Adicione mais categorias conforme necessário
-        ],
-      ),
-    );
-  }
-
-  Widget _buildCategoryListView(String category, List<MovieItem> items) {
-    return Column(
-      crossAxisAlignment: CrossAxisAlignment.start,
-      children: [
-        Padding(
-          padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
-          child: Text(category, style: const TextStyle(fontSize: 20, fontWeight: FontWeight.bold),),
-        ),
-        Container(
-          height: 250,
-          child: ListView.builder(
-            scrollDirection: Axis.horizontal,
-            itemCount: items.length,
-            itemBuilder: (BuildContext context, int index) {
-              final item = items[index];
-              return GestureDetector(
-                onTap: () {
-                  Navigator.push(
-                    context,
-                    MaterialPageRoute(
-                      builder: (context) => MovieDetailsView(movie: item),
+  Widget _buildBody(
```

Comment:
Evite usar helper methods pra renderizar widgets, isso pode custar muuuito no desempenho do seu app. Prefira criar widgets separados.
Segue o vídeo do time do Google explicando isso: https://youtu.be/IOyq-eTRhvo?si=eLnvexmlXwI37dhe